src/v2/manifest/mod.rs

```diff
@@ -141,6 +105,63 @@ impl Client {
         }
     }
 
+    async fn build_url(&self, name: &str, reference: &str) -> Result<Url> {
```

Comment:
I think this doesn't need to be `async`.

Reply:
It does, otherwise `get_manifest_and_ref` call explodes, saying that Result doesn't implement Future

Reply:
Sorry, my comment was not explicit enough.

You can drop the `async` here and at the same time also drop the matching `.await` when you call `build_url`. That is, `fn build_url` coupled with `self.build_url(x, y)?`.

Reply:
Oh, I trusted RLS way too much as it had a hiccup and threw and error initially. It does work now, pushed. Thanks!